pkg/deployments/manager.go

```diff
@@ -109,6 +109,18 @@ func (r *Manager) getScaler(deploymentName string) *scaler {
 	return b
 }
 
+// getScalesSnapshot returns a snapshot of the stats for all scalers managed by the Manager.
+// The scales are returned as a map, where the keys are the model names
+func (r *Manager) getScalesSnapshot() map[string]scale {
+	r.scalersMtx.Lock()
```

Comment:
fyi: a `sync.Map` would be a better fit instead of locking via `scalersMtx`

Reply:
I sometimes wonder about this, but tend to agree with the comment in the docs on `sync.Map` about maintaining type safety:

> Most code should use a plain Go map instead, with separate locking or coordination, for better type safety and to make it easier to maintain other invariants along with the map content.

https://pkg.go.dev/sync#Map

Reply:
Type safety is an argument. I will close the other PRs and may do another spike to propose a generic `SafeGetOrCreate(map, rwLock, key, factoryMethod)T` method. Write locks that can be triggered by external requests should be avoided